hail/python/hail/expr/type_parsing.py

```diff
@@ -5,8 +5,8 @@
 
 from . import types
 
-type_grammar = Grammar(r"""
-    type = _ (array / ndarray / set / dict / struct / union / tuple / interval / int64 / int32 / float32 / float64 / bool / str / call / str / locus / void / variable) _
+type_grammar_str = r"""
+    type = _ ( array / bool / call / dict / interval / int64 / int32 / float32 / float64 / locus / ndarray / set / stream / struct / str / tuple / union / void / variable / rng_state ) _
```

Comment:
On the one hand, if you're going to sort these, why not go all the way and sort the declarations below?
On the other hand, it's more churn.
Up to you.

Reply:
These are partially sorted because alternations in the grammar must be prefix free. I'll move `rng_state` though, I think I got lazy.